stats_test.go

```diff
@@ -0,0 +1,131 @@
+package codeowners
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCalculateOwnershipStats(t *testing.T) {
```

Comment:
I'll be honest, Augment wrote most of this. That said: I did see these tests fail, and they helped me get my "total owners" count to work correctly when 100% and 0% of the files are owned.

Let me know if I'm doing anything that looks wrong, and if you can think of any other test cases you'd like to see.